rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/importexport/Sw360ImportExportService.java

```diff
@@ -244,5 +279,85 @@ public void getComponentDetailedExport(User sw360User, HttpServletResponse respo
         response.setHeader(CONTENT_DISPOSITION, String.format("Components; filename=\"%s\"", filename));
         FileCopyUtils.copy(byteArrayInputStream, response.getOutputStream());
+
+    }
+
+    @JsonInclude
+    public RequestSummary uploadComponent(User sw360User, MultipartFile file, HttpServletRequest request,
+            HttpServletResponse response) throws IOException, TException, ServletException {
+        if (!PermissionUtils.isUserAtLeast(UserGroup.ADMIN, sw360User)) {
+            throw new AccessDeniedException("Unable to upload component csv file. User is not admin");
+        }
+        List<CSVRecord> releaseRecords = getCSVFromRequest(request, "file");
+        FluentIterable<ComponentCSVRecord> compCSVRecords = convertCSVRecordsToCompCSVRecords(releaseRecords);
+        ComponentService.Iface sw360ComponentClient = getThriftComponentClient();
+        VendorService.Iface sw360VendorClient = getThriftVendorClient();
+        AttachmentService.Iface sw360AttachmentClient = getThriftAttachmentClient();
+        RequestSummary requestSummary = writeToDatabase(compCSVRecords, sw360ComponentClient, sw360VendorClient,
+                sw360AttachmentClient, sw360User);
+        return requestSummary;
+    }
+
+    private List<CSVRecord> getCSVFromRequest(HttpServletRequest request, String fileUploadFormId)
+            throws IOException, TException, ServletException {
+        final InputStream stream = getInputStreamFromRequest(request, fileUploadFormId);
+        return readAsCSVRecords(stream);
+    }
+
+    private InputStream getInputStreamFromRequest(HttpServletRequest request, String fileUploadFormId)
+            throws IOException, ServletException {
+        Collection<Part> parts = request.getParts();
+
+        for (Part part : parts) {
+            if (!part.getName().equals(fileUploadFormId)) {
+                return part.getInputStream();
+            }
+        }
+        throw new IOException("File not found in the request with the specified field name.");
+    }
+
+    private ComponentService.Iface getThriftComponentClient() throws TTransportException {
+        THttpClient thriftClient = new THttpClient(thriftServerUrl + "/components/thrift");
+        TProtocol protocol = new TCompactProtocol(thriftClient);
+        return new ComponentService.Client(protocol);
+    }
+
+    private VendorService.Iface getThriftVendorClient() throws TTransportException {
+        THttpClient thriftClient = new THttpClient(thriftServerUrl + "/vendors/thrift");
+        TProtocol protocol = new TCompactProtocol(thriftClient);
+        return new VendorService.Client(protocol);
+    }
+
+    private AttachmentService.Iface getThriftAttachmentClient() throws TTransportException {
+        THttpClient thriftClient = new THttpClient(thriftServerUrl + "/attachments/thrift");
+        TProtocol protocol = new TCompactProtocol(thriftClient);
+        return new AttachmentService.Client(protocol);
+    }
```

Comment:
The functions are already defined in `libraries/datahandler/src/main/java/org/eclipse/sw360/datahandler/thrift/ThriftClients.java`. Please use them instead.